src/simplewallet/simplewallet.cpp

```diff
@@ -3600,14 +3614,31 @@ bool simple_wallet::print_ring_members(const std::vector<tools::wallet2::pending
   return true;
 }
 //----------------------------------------------------------------------------------------------------
-bool simple_wallet::transfer_main(int transfer_type, const std::vector<std::string> &args_)
+bool simple_wallet::transfer_main(int transfer_type, const std::vector<std::string> &args_, size_t try_lowest_priority)
 {
 //  "transfer [index=<N1>[,<N2>,...]] [<priority>] [<ring_size>] <address> <amount> [<payment_id>]"
-  if (m_wallet->ask_password() && !get_and_verify_password()) { return true; }
+  if (try_lowest_priority != 2 && m_wallet->ask_password() && !get_and_verify_password()) { return true; }
   if (!try_connect_to_daemon())
     return true;
 
-  LOCK_IDLE_SCOPE();
+  // need to tweak LOCK_IDLE_SCOPE() so that it doesn't get invoked when try_lowest_priority==2
```

Comment:
Why ?

Reply:
If `LOCK_IDLE_SCOPE();` is kept here, it'll be invoked for the second time when `transfer_main` is called recursively with `try_lowest_priority` being 2, which causes deadlock.

Reply:
Does it need to be recursive in the first place ?

Reply:
Well, this is the least intrusive way I could come up with. Can you suggest an alternative approach?

Reply:
I'd do the txpool check at the top of transfer_main, if the priority is 0 (default). Then, depending on whether you estimate there's backlog, set priority to 1 or 2.